qcodes/instrument_drivers/tektronix/DPO7200xx.py

```diff
@@ -0,0 +1,605 @@
+            constant as you change the time per division 
+            setting. Record length is read only.
+
+            Constant mode attempts to keep sample rate 
+            constant as you change the time per division 
+            setting. Record length is read only.
+
+            Manual mode lets you change sample mode and 
+            record length. Time per division or Horizontal 
+            scale is read only.
+            """
+        )
+
+        self.add_parameter(
+            "unit",
+            get_cmd="HORizontal:MAIn:UNIts?",
+            get_parser=strip_quotes
+        )
+
+        self.add_parameter(
+            "record_length",
+            get_cmd="HORizontal:MODE:RECOrdlength?",
+            set_cmd=self._set_record_length,
+            get_parser=float
+        )
+
+        self.add_parameter(
+            "sample_rate",
+            get_cmd="HORizontal:MODE:SAMPLERate?",
+            set_cmd="HORizontal:MODE:SAMPLERate {}",
+            get_parser=float,
+            unit=f"sample/{self.unit()}"
+        )
+
+        self.add_parameter(
+            "scale",
+            get_cmd="HORizontal:MODE:SCAle?",
+            set_cmd=self._set_scale,
+            get_parser=float,
+            unit=f"{self.unit()}/div"
+        )
+
+        self.add_parameter(
+            "position",
+            get_cmd="HORizontal:POSition?",
+            set_cmd="HORizontal:POSition {}",
+            get_parser=float,
+            unit="%",
+            docstring="""
+            The horizontal position relative to a 
+            received trigger. E.g. a value of '10'
+            sets the trigger position of the waveform 
+            such that 10% of the display is to the 
+            left of the trigger position.
+            """
+        )
+
+        self.add_parameter(
+            "roll",
+            get_cmd="HORizontal:ROLL?",
+            set_cmd="HORizontal:ROLL {}",
+            vals=Enum("Auto", "On", "Off"),
+            docstring="""
+            Use Roll Mode when you want to view data at 
+            very slow sweep speeds.
+            """
+        )
+
+    def _set_record_length(self, value: int) -> None:
+        if self.mode() != "manual":
+            raise ModeError(
+                "The record length can only be changed in manual mode"
+            )
+
+        self.write(f"HORizontal:MODE:RECOrdlength {value}")
+
+    def _set_scale(self, value):
+        if self.mode() == "manual":
+            raise ModeError(
+                "The scale cannot be changed in manual mode"
+            )
+
+        self.write(f"HORizontal:MODE:SCAle {value}")
+
+
+class TektronixDPOMeasurement(InstrumentChannel):
+    """
+    The measurement submodule
+    """
+    # It was found by trial and error that adjusting
+    # the measurement type and source takes some time
+    # to reflect properly on the value of the
+    # measurement. Wait a minimum of ...
+    minimum_adjustment_time = 0.1
```

Comment:
should this be private ? :) and explained in the example notebook as well?

Reply:
I made this private now